actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -50,7 +50,14 @@ def authenticated_encrypted_cookie_salt
     end
 
     def secret_token
-      get_header Cookies::SECRET_TOKEN
+      secret_token = get_header Cookies::SECRET_TOKEN
+      if secret_token
+        ActiveSupport::Deprecation.warn(<<-MSG.squish)
+         cookies.secret_token is deprecated and will be removed in Rails 6.0.
+         Please use cookies.secret_key_base instead.
+        MSG
+      end
+      secret_token
```

Comment:
This is an internal method (note the `# :stopdoc:` above), so we don't need this deprecation warning or the Action Pack changelog entry.